src/Tests.Microsoft.Security.Utilities.Core/SecretMaskerTests.cs

```diff
@@ -755,6 +755,35 @@ public void SecretMasker_DistinguishLiteralAndRegexRedactionTokens()
         Assert.AreEqual("zzzxyyyab12", result);
     }
 
+    [TestMethod]
+    public void SecretMasker_ProvidesDetectionsViaCallback()
```

Comment:
>SecretMasker_ProvidesDetectionsViaCallback [](http://example.com/codeflow?start=16&length=42)

Related to other discussion. You should consider updating this test (or adding a new one) that provokes both contiguous and overlapping matches (because in our implementation we still require the code to fire a callback for every individual match.

The regexes below would do well for this

'patter'
'pattern'
'n1'

with data like 

'yada patter pattern pattern1'

The masker should collapse this to
'yada +++ +++ +++'

But we should get six callbacks?

Reply:
Will do

Reply:
Hmm, there's an annoying edge case here.

If detections overlap and have the same start index, then which redaction token is supposed to win? I think before my change, it was whichever detection came first, which is actually not well-defined either because it relies on hashset enumeration order. With my change, I added an unstable sort to the mix. Blah.

Maybe we should be sorting by redaction token too? I'll do that.

Reply:
Outstanding feedback here because I broke this when I optimized how we process the list of detections. Fixed. 😊